tests/test_app/test_init.py

```diff
@@ -181,6 +174,7 @@ def main(self, val: int) -> int:
         return val + self.constant
 
 
+@pytest.mark.xfail(reason="Constructing apps on the fly is no longer supported")
```

Comment:
This comment not really directed at this test, but your comment raises an interesting issue. I think we should still support constructing apps on the fly, BUT not support them being hooked into the extension system. So they would have to be explicitly imported.

Reply:
Now to address this xfail. We do need to be testing that, for a given app, we correctly generate docs. IO think this means we should write a new function called `_make_apphelp_docstring(app)`. This should basically execute the commands within `app_help()`
```python
def _make_apphelp_docstring(app):
    app_doc = app.__doc__ or ""
    init_doc = app.__init__.__doc__ or ""
    if init_doc.strip():
        docs.extend(["", _clean_params_docs(init_doc)])

    types = _make_types(app)
    docs.extend([""] + _make_head("Input type") + [", ".join(types["_data_types"])])
    docs.extend([""] + _make_head("Output type") + [", ".join(types["_return_types"])])

    return "\n".join(docs)

def app_help(...):
    ...
    print(_make_apphelp_docstring(app))
```
This function can be tested on any app.